ql/src/java/org/apache/hadoop/hive/ql/ddl/misc/insert/commit/hook/InsertCommitHookOperation.java

```diff
@@ -16,7 +16,7 @@
  * limitations under the License.
  */
 
-package org.apache.hadoop.hive.ql.ddl.misc;
+package org.apache.hadoop.hive.ql.ddl.misc.insert.commit.hook;
```

Comment:
Maybe all these could stay under misc package. I am not sure we need so many individual packages with single classes at this level. Or at least we could have more general packages, e.g., hooks, metadata. What do you think?

Reply:
I think that it is a good structure that each operation have their own package, it makes things cleaner. Putting everything under misc would break this structure, and also has a potential problem if in the future there would be more operations under misc, it may be a huge package with an endless list of classes.

I'm ok with renaming the sub packages:

cache.metadata -> metadata
insert.commit.hook -> hooks
msck -> msck
remove.flag -> flags (?)
show.conf -> conf

Would that be ok? I don't see which of these operations are related enough to be put under the same category, so we'll still have 5 packages, but if in the future we'll have some new configuration, or metadata, or hook related stuff, they can go to the same package, or under some sub packages.